unified-runtime/source/adapters/level_zero/v2/command_buffer.cpp

```diff
@@ -97,6 +121,17 @@ ur_result_t ur_exp_command_buffer_handle_t_::finalizeCommandBuffer() {
   // It is not allowed to append to command list from multiple threads.
   auto commandListLocked = commandListManager.lock();
   UR_ASSERT(!isFinalized, UR_RESULT_ERROR_INVALID_OPERATION);
+
+  if (!isInOrder) {
+    ZE2UR_CALL(zeCommandListAppendBarrier,
+               (commandListLocked->getZeCommandList(), nullptr, 0, nullptr));
+    for (auto &event : syncPoints) {
+      ZE2UR_CALL(zeCommandListAppendEventReset,
+                 (commandListLocked->getZeCommandList(), event->getZeEvent()));
+    }
+    ZE2UR_CALL(zeCommandListAppendBarrier,
+               (commandListLocked->getZeCommandList(), nullptr, 0, nullptr));
+  }
```

Comment:
It's worth noting in the v1 adapter that we made a separate L0 command-list just for the `zeCommandListAppendEventReset` commands. Then scheduled the reset command-list after the main command-list as a dependency, which meant we could signal back to the user that the graph was complete when the main command-list completed, without waiting on the event reset command-list as part of that.

That was the theory anyway, as I have no numbers to backup any performance benefit of that approach. Happy to keep this PR as is with this simpler and easier to maintain approach.